src/scheduler/node_info.cpp

```diff
@@ -491,16 +492,12 @@ query_node_info(struct batch_status *node, server_info *sinfo)
 		else if (!strcmp(attrp->name, ATTR_NODE_jobs))
 			ninfo->jobs = break_comma_list(attrp->value);
 		else if (!strcmp(attrp->name, ATTR_msvr_remote_jobs)) {
-			char *ptr = NULL;
-			char *jidptr = attrp->value;
-
-			for (ptr = strchr(attrp->value, ','); ptr != NULL; jidptr = ptr + 1, ptr = strchr(ptr + 1, ',')) {
-				*ptr = '\0';
-				ninfo->msvr_rmt_jobs.insert(jidptr);
-				*ptr = ',';
+			std::string rmtjid;
+			std::istringstream ss(attrp->value);
+			while (ss) {
```

Comment:
you will have to loop on return value of getline instead. I tried what you did and that seems to count the last job twice.
some thing like - 
while(std::getline(ss, rmtjid, ',')) {
    ninfo->msvr_rm_jobs.insert(rmtjid);
}

Reply:
thanks for trying this out and catching an issue!